src/solvers/flattening/arrays.cpp

```diff
@@ -547,12 +531,9 @@ void arrayst::add_array_constraints_with(
   // use other array index applications for "else" case
   // add constraint x[I]=y[I] for I!=i
 
-  for(index_sett::const_iterator
-      it=index_set.begin();
-      it!=index_set.end();
-      it++)
+  for(const auto & it : index_set)
```

Comment:
See above (not an iterator, space); specifically use `other_index` to avoid the useless assignment below.